### PR TITLE
Add wire render options to wire.create

### PR DESCRIPTION
**Pull request by ajloveslily14**
I made sure they're optional so old scripts don't break

### CODE REVIEW

**Comment by thegrb93** [comment]
Some things to improve this,

change ValidWireMat to the format {[material_name] = true, ...} and remove table.HasValue
put the list of valid materials in the material parameter doc
change
```
local width = checkluatype(width, TYPE_NUMBER, 0, 0)
local color = checktype(color, SF.Types['Color'], 0, SF.Color.Wrap(Color(255,255,255)))
local material = ValidWireMat[material] and material or "cable/rope"
```

Also I'm not sure if the wire width needs to be clamped. Does it?

**Comment by ajloveslily14** [comment]
Made the formatting, I am not sure how to add the valid materials to the documentation as there doesn't seem to be any other place that something like that is done, so I havn't done it.

**Comment by thegrb93** [comment]
You just put them in the comment with the parameter

**Comment by ajloveslily14** [comment]
alright, I think I did it

**Comment by thegrb93** [comment]
Can you make the default color white and put spaces after your commas? after that, I think it's good and @Damianu can merge once its tested

File: lua/starfall/libs_sv/wire.lua [hunk 1]
```diff
@@ -318,17 +318,29 @@ function wire_library.serverUUID()
...
 	checktype(entO, SF.Types["Entity"])
 	checkluatype(inputname, TYPE_STRING)
 	checkluatype(outputname, TYPE_STRING)
-
+	
+	local width = width or 0
+	checkluatype(width, TYPE_NUMBER,0,0)
+	width = math.Clamp(width,0,5)
+	
+	checktype(color, SF.Types['Color'],0,SF.Color.Wrap(Color(255,0,0)))
```
Comment (Damianu) [inline lua/starfall/libs_sv/wire.lua]: It returns default val but u never assign it. Also keep it consistent with other libs and just split it to:
```
color = color or SF.Color.Wrap(Color(255, 255, 255))
checktype(color, SF.Types['Color']) ```

Comment (thegrb93) [reply lua/starfall/libs_sv/wire.lua]: That's less efficient.

**Comment by thegrb93** [comment]
You still need to do
```
local width = checkluatype(width, TYPE_NUMBER, 0, 0)
local color = checktype(color, SF.Types['Color'], 0, SF.Color.Wrap(Color(255,255,255)))
local material = ValidWireMat[material] and material or "cable/rope"
```

**Comment by ajloveslily14** [comment]
I think with that fix it's good to go right? Unless you also mean remove the return lines between the checks

**Pull request merged**